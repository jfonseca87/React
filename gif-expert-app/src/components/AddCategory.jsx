import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategory }) => {
  const [inputValue, setInputValue] = useState();

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const newValue = inputValue.trim();
    if (newValue.length <= 2) return;

    setCategory(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}