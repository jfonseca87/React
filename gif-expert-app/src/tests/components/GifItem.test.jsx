import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe("test on <GifItem />", () => {
    const title = "Goku";
    const url = "https://algo/goku.jpg"

    test("should made match with the snapshot",  () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })

    test("should image has source and alt properties", () => {
        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole("img");
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })

    test("should has title", () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});