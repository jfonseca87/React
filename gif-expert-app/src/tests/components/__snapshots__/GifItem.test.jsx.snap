// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifItem /> should made match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://algo/goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
